compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -802,6 +811,12 @@ convertExpr env0 e = do
         t2' <- convertType env t2
         pure (x' `ETyApp` t1' `ETyApp` t2' `ETmApp` EBuiltin (BEText (unpackCStringUtf8 s)))
 
+    go env (ConstraintTupleProjection i j) args | (LExpr x : args') <- drop j args =
+        fmap (, args') $ do
+            let fieldName = mkField ("_" <> T.pack (show i))
```

Comment:
Should we factor out the naming scheme used for this fields so that we can make sure that this conversion and the definition of the struct type agree?

Reply:
That sounds good. It shows up in a few more places as well, so I made it a utility function.